include/deal.II/base/mpi_tags.h

```diff
@@ -0,0 +1,99 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_mpi_tags_h
+#define dealii_mpi_tags_h
+
+#include <deal.II/base/config.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace Utilities
+{
+  namespace MPI
+  {
+    namespace internal
+    {
+      /**
+       * This enum holds all MPI tags used in collective MPI communications with
+       * using MPI_ANY_SOURCE inside the deal.II library.
```

Comment:
Shouldn't we, in principle, use this approach for *all* tags?

Reply:
Good question. I guess it could be possible that one of the tags used here collide with another communication somewhere else? I will check what other places I can find.